src/rotten-tomatoes/index.js

```diff
@@ -1,3 +1,123 @@
+    queue.push(flattened)
+
+    //  Use Breadth First Search function to count minutes, traverse and update the grid
+    const result = BFS(count, adjacentsArr, queue, grid, visited)
+    count += result[0]
+    grid = result[1]
+
+    //  Return -1 if there is any tomato that will never be rotten ie, shielded/immune tomatoe exist.
+    for (let row = 0; row < grid.length; row++){
+        for (let column = 0; column < grid[row].length; column++){
+            if (grid[row][column] === 1){
+                let immuneAdjacents = 0
+                for (const position of adjacentsArr[row][column]){
+                    if (grid[position[0]][position[1]] === 0 || grid[position[0]][position[1]] === 1){
+                        immuneAdjacents++
+                    }
+                }
+                if (immuneAdjacents === adjacentsArr[row][column].length){
+                    return -1
+                }
+            }
+        }
+    }
+    return count;
+}
+
+// Helpers Functions
+
+//  Function to fetch the adjacent positions of each tomato in the grid
+function adjacent(rowLength, columnLength, position){
+    const top = position[0]===0 ? false : [position[0]-1, position[1]]
+    const bottom = position[0]===rowLength ? false : [position[0]+1, position[1]]
+    const right = position[1]===columnLength ? false : [position[0], position[1]+1]
+    const left = position[1]===0 ? false : [position[0], position[1]-1]
+    
+    return [right, left, bottom, top].filter(Boolean);
+}
+
+//  Breadth First Search function to count minutes, traverse and update the grid
+function BFS(count, allAdjacents, queue, grid, visited){
+
+    //  Queue up the adjacent positions of the current level being treated
+    let adjacentArr = queue[0]
+    nextLevel = []
+    let checkCount = 0
+    for (const position of adjacentArr){
+        if (grid[position[0]][position[1]] === 1 && isVisited(position, visited) === false){
+            checkCount++
+            grid[position[0]][position[1]] = 2
+            nextLevel.push(allAdjacents[position[0]][position[1]])
+        }
+    }
+    nextLevel = nextLevel.flat()
+    queue.push(nextLevel)
+
+    //  Update visited positions and remove the position you have attended to from the queue
+    const shifted = queue.shift()
+    for (const position of shifted){
+        if (isVisited(position, visited) === false)
+            visited.push(position)
+    }
+    if (checkCount > 0)
+        count++
+
+    //  Termination point of this recursive BFS function
+    if (queue.flat().length === 0){
+        return [count, grid]
+    }
+    
+    //  Repeat the process if termination condition is yet to be met
+    let result = BFS(count, allAdjacents, queue, grid, visited)
+    return result;
+}
+
+//  Function to check if a position has been visited
+function isVisited(position, visitedPositions){
+    for (const spot of visitedPositions){
+        if (spot.toString() === position.toString())
```

Comment:
Why do they both need to be strings to be compared?

Reply:
Because they are both arrays and when I tried comparing directly initially, it wasn't working.

Reply:
That's because you can't compare two array values.

```js
[] === [] // false
```

Likewise,
```js
[1] === [1] // false
```

What you're comparing here is not the value(s) of the array but the reference to them, which are different. If you've seen pointers, explain in CS50, you'd have a better understanding.

So, what you did here is a superficial check. If you want to compare that the two values are arrays though, regardless of their value, then use `Array.isArray()`

Reply:
I wasn't trying to compare if it is an array, sir, I was trying to compare the values in the arrays (which is a position in the grid) if they are both pointing to the same position in the grid.

Reply:
Then, you can't do that with the equality sign.

Reply:
Yes sir. that is why I used `toString()`.  Please what should I have done instead of using `toString()`?

Reply:
`.toString()` is wrong. You have to iterate over the array to check their values.

Reply:
Noted sir. Thank you.